bindings/java: implement stream binding methods (int, InputStream, int) in JDBC4PreparedStatement

moonwhistle opened the pull request:
## Purpose

* Implement `setAsciiStream(int, InputStream, int)`, `setUnicodeStream(int, InputStream, int)`, and `setBinaryStream(int, InputStream, int)` methods in JDBC4PreparedStatemen

## Changes
* `setAsciiStream(int, InputStream, int)`: Reads ASCII bytes, converts to `String` using `US_ASCII` and binds with `bindText()`.

* `setUnicodeStream(int, InputStream, int)`: Reads bytes as `UTF-8` encoded text and binds with `bindText()`.

* `setBinaryStream(int, InputStream, int)`: Reads raw bytes and binds with `bindBlob()`.

* Added consistent error handling and validation
  * null stream - `bindNull()`
  * Negative length - throws `SQLException`
  * Empty stream  - Empty String or Empty Array
  * I/O errors - throw `SQLException`

* Ensures consistency between `setXxxStream` and `getXxxStream` methods, so data written and read use the same encoding.

## Related Issue
* #615

seonWKim commented on bindings/java/src/main/java/tech/turso/jdbc4/JDBC4PreparedStatement.java:
```diff
@@ -164,17 +167,96 @@ public void setTimestamp(int parameterIndex, Timestamp x) throws SQLException {
+          break;
+        }
+      }
+      byte[] data = baos.toByteArray();
```
Btw `toByteArray()` also internally creates a copy. But I think this is the good enough for now.

moonwhistle replied:
I’ll look into better options for reducing copy overhead later and see if there’s a more efficient approach available.

moonwhistle commented on bindings/java/src/main/java/tech/turso/jdbc4/JDBC4PreparedStatement.java:
```diff
@@ -164,17 +167,96 @@ public void setTimestamp(int parameterIndex, Timestamp x) throws SQLException {
+      byte[] buffer = new byte[8192];
+      int bytesRead;
+      int totalRead = 0;
+      while ((bytesRead = x.read(buffer, 0, Math.min(buffer.length, length - totalRead))) != -1) {
```
I totally missed that point.

First of all, used `totalRead < length` to make sure it doesn’t read beyond the given length.

And I changed the condition to `(bytesRead = x.read(buffer, 0, Math.min(buffer.length, length - totalRead))) > 0`,
so even if it reads `0`, the loop stops properly instead of running indefinitely.

moonwhistle commented:
In `setAsciiStream` and `setUnicodeStream`

As-is

```
while (offset < length) {
  int readBytes = x.read(buffer, offset, length - offset);
  if (readBytes == -1) {
    break;
  }
  offset += readBytes;
}
```

To-be
```
int read;
while (offset < length && (read = x.read(buffer, offset, length - offset)) > 0) {
  offset += read;
}
```

I changed this to prevent a potential infinite loop when read() `returns 0`while data still exists.
Also added an early return when `length == 0`.

seonWKim reviewed (approved):
LGTM
